src/pages/workspace/companyCards/WorkspaceCompanyCardsList.tsx

```diff
@@ -1,69 +1,120 @@
-import React, {useCallback, useMemo} from 'react';
-import type {ListRenderItemInfo} from 'react-native';
-import {FlatList, View} from 'react-native';
+import type {FlashListRef, ListRenderItemInfo} from '@shopify/flash-list';
+import {FlashList} from '@shopify/flash-list';
+import React, {useCallback, useMemo, useRef} from 'react';
+import {View} from 'react-native';
 import type {OnyxEntry} from 'react-native-onyx';
 import OfflineWithFeedback from '@components/OfflineWithFeedback';
 import {PressableWithFeedback} from '@components/Pressable';
 import SearchBar from '@components/SearchBar';
 import Text from '@components/Text';
+import useCardFeeds from '@hooks/useCardFeeds';
+import {useCompanyCardFeedIcons} from '@hooks/useCompanyCardIcons';
 import useLocalize from '@hooks/useLocalize';
 import useOnyx from '@hooks/useOnyx';
 import usePolicy from '@hooks/usePolicy';
+import useResponsiveLayout from '@hooks/useResponsiveLayout';
 import useSearchResults from '@hooks/useSearchResults';
+import useThemeIllustrations from '@hooks/useThemeIllustrations';
 import useThemeStyles from '@hooks/useThemeStyles';
-import {filterCardsByPersonalDetails, getCardsByCardholderName, getCompanyCardFeedWithDomainID, getDefaultCardName, sortCardsByCardholderName} from '@libs/CardUtils';
+import {
+    filterCardsByPersonalDetails,
+    getCardFeedIcon,
+    getCardsByCardholderName,
+    getCompanyCardFeedWithDomainID,
+    getCompanyFeeds,
+    getPlaidInstitutionIconUrl,
+    sortCardsByCardholderName,
+} from '@libs/CardUtils';
 import {getMemberAccountIDsForWorkspace} from '@libs/PolicyUtils';
 import Navigation from '@navigation/Navigation';
 import CONST from '@src/CONST';
 import ONYXKEYS from '@src/ONYXKEYS';
 import ROUTES from '@src/ROUTES';
-import type {Card, CompanyCardFeed, WorkspaceCardsList} from '@src/types/onyx';
+import type {Card, CompanyCardFeed, CompanyCardFeedWithDomainID, WorkspaceCardsList} from '@src/types/onyx';
+import WorkspaceCompanyCardsListRow from './WorkspaceCompanyCardListRow';
 import WorkspaceCompanyCardsFeedAddedEmptyPage from './WorkspaceCompanyCardsFeedAddedEmptyPage';
-import WorkspaceCompanyCardsListRow from './WorkspaceCompanyCardsListRow';
 
 type WorkspaceCompanyCardsListProps = {
+    /** Selected feed */
+    selectedFeed: CompanyCardFeedWithDomainID;
+
     /** List of company cards */
     cardsList: OnyxEntry<WorkspaceCardsList>;
 
     /** Current policy id */
     policyID: string;
 
     /** Handle assign card action */
-    handleAssignCard: () => void;
+    onAssignCard: () => void;
 
     /** Whether to disable assign card button */
-    isDisabledAssignCardButton?: boolean;
+    isAssigningCardDisabled?: boolean;
+
+    shouldShowAssignCardButton?: boolean;
```

Comment:
Add a comment explaining this one

Reply:
Removed this, i don't think we need it anymore as this was related to showing the "global" "Assign cards" button